Implement books scraper with Scrapy

vladyslav-tmf opened the pull request:
- Create Scrapy spider for books.toscrape.com
- Add BookItem model with fields for book details
- Implement parsing logic for book information
- Add progress bar with tqdm
- Configure logging level to reduce output verbosity
- Enable HTTP caching for better performance
- Replace requirements.txt with pyproject.toml for Poetry dependency management
<details>
<summary>Result:</summary>

![Screenshot_1](https://github.com/user-attachments/assets/6be3a590-e31b-4899-9a48-9b955f3409de)

</details>

vladyslav-tmf committed:
Implement books scraper with Scrapy

- Create Scrapy spider for books.toscrape.com
- Add BookItem model with fields for book details
- Implement parsing logic for book information
- Add progress bar with tqdm
- Configure logging level to reduce output verbosity
- Enable HTTP caching for better performance
- Replace requirements.txt with pyproject.toml for Poetry dependency management

mateacademy-ai-mentor commented on .gitignore:
```diff
@@ -6,3 +6,5 @@
 venv/
 .pytest_cache/
 **__pycache__/
+.scrapy
```
There is an extra space after `.scrapy`. This might cause the directory to not be ignored properly. Consider removing the space to ensure `.scrapy` is correctly ignored.

mateacademy-ai-mentor commented on books_scraper/spiders/books.py:
```diff
@@ -0,0 +1,83 @@
+            response.css("p.price_color::text").get().replace("£", "")
+        )
+
+        stock_text = (
+            response.css("p.instock.availability::text").getall()[1].strip()
+        )
+        book["amount_in_stock"] = int(stock_text.split("(")[1].split()[0])
```
The code assumes that the stock text is always in the format 'In stock (xx available)'. If the format changes or if the text is not present, this could raise an error. Consider adding error handling to manage unexpected formats or missing data.

mateacademy-ai-mentor reviewed (approved):
Great job on implementing the Scrapy project! 🎉 Your code meets the task requirements and is well-structured. There are a couple of minor issues to address: consider removing the extra space after `.scrapy` in the `.gitignore` file to ensure it's ignored properly, and add error handling for unexpected stock text formats in the `BooksSpider` class. Keep up the good work and happy coding! 😊

✨ Please rate this review in the Platform's notification center. I will be grateful for your feedback! ✨

vladyslav-tmf committed:
Refactor code

- Convert BookItem to dataclass for better data structure
- Create book instance using named parameters

Arsen-hrynevych reviewed (approved):
ok
